src/model/line/parser.rs

```diff
@@ -42,7 +42,7 @@ impl Note {
     }
 }
 
-pub fn parse_line(line: &str) -> Result<Line> {
+pub fn parse(line: &str) -> Result<Line> {
```

Comment:
I think it'd be nice to split the parser itself out from this `parse` function, so that it could be combined into a bigger parser. (I can image a file with a `Line` per line of text.)

Reply:
Ok I restructured so that I kept this line-parsing stuff as a separate file/module but made it use the same `parser()`/`parse()` pattern as eg `Progression` (which sets it up to also be included if we do refactor that into a trait)